projects/core/src/configurator/commons/facade/configurator-groups.service.ts

```diff
@@ -135,6 +97,33 @@ export class ConfiguratorGroupsService {
     );
   }
 
+  /**
+   * Determines the group status by the group ID and the switcher that defines whether the group has been visited or not.
+   *
+   * @param owner - Owner
+   * @param groupId - Group ID
+   * @param setGroupVisited - Determines whether the group has to be set as visited or not
+   */
+  public setGroupStatus(
+    owner: GenericConfigurator.Owner,
+    groupId: string,
+    setGroupVisited: Boolean
+  ) {
+    this.configuratorCommonsService
+      .getConfiguration(owner)
+      .pipe(
+        map((configuration) =>
+          this.configuratorGroupStatusService.setGroupStatus(
+            configuration,
+            groupId,
+            setGroupVisited
+          )
+        ),
+        take(1)
+      )
+      .subscribe();
```

Comment:
Why do we need to take(1) and subscribe?

Reply:
We need to read the Configuration to call setGroupStatus of the group status service.
This should not be done on each change of the configuration (set group status itself is a change of the configuration), which would result in an endless loop.

I could move the coding from the "map" method to the subscribe method, maybe that would make it more clear.